sources/packages/api/src/helpers/utilfunctions.ts

```diff
@@ -1,3 +1,110 @@
+
 export default class Helper {
-  //Placeholder for Helper methods
+  /**
+   * helper method to map attributes from designation agreement location model to institution location dto.
+   * @param designationAgreementLocationData
+   * @returns designation agreement locations and their designation statuses.
+   */
+  static mapDesignationAgreementLocationToInstitutionLocationDTO(
+    designationAgreementLocationData: DesignationAgreementLocation[],
+  ) {
+    return designationAgreementLocationData.map(
+      (el: DesignationAgreementLocation) => {
+        return {
+          id: el.institutionLocation.id,
+          name: el.institutionLocation.name,
+          designationStatus: this.getDesignationStatus(
+            el.designationAgreement.designationStatus,
+            el.approved,
+            el.designationAgreement.startDate,
+            el.designationAgreement.endDate,
+          ),
+          data: {
+            address: {
+              addressLine1: el.institutionLocation.data.address?.addressLine1,
+              addressLine2: el.institutionLocation.data.address?.addressLine2,
+              province: el.institutionLocation.data.address?.province,
+              country: el.institutionLocation.data.address?.country,
+              city: el.institutionLocation.data.address?.city,
+              postalCode: el.institutionLocation.data.address?.postalCode,
+            },
+          },
+          primaryContact: {
+            primaryContactFirstName:
+              el.institutionLocation.primaryContact?.firstName,
+            primaryContactLastName:
+              el.institutionLocation.primaryContact?.lastName,
+            primaryContactEmail: el.institutionLocation.primaryContact?.email,
+            primaryContactPhone:
+              el.institutionLocation.primaryContact?.phoneNumber,
+          },
+          institution: {
+            institutionPrimaryContact: {
+              primaryContactEmail:
+                el.institutionLocation.institution.institutionPrimaryContact
+                  .primaryContactEmail,
+              primaryContactFirstName:
+                el.institutionLocation.institution.institutionPrimaryContact
+                  .primaryContactFirstName,
+              primaryContactLastName:
+                el.institutionLocation.institution.institutionPrimaryContact
+                  .primaryContactLastName,
+              primaryContactPhone:
+                el.institutionLocation.institution.institutionPrimaryContact
+                  .primaryContactPhone,
+            },
+          },
+          institutionCode: el.institutionLocation.institutionCode,
+        } as InstitutionLocationsDetailsDto;
+      },
+    );
+  }
+
+  /**
+   * internal helper function to check if location is designated or not.
+   * @param  designationAgreementStatus
+   * @param  isApproved
+   * @param  startDate
+   * @param  endDate
+   * @returns designation status.
+   */
+  private static getDesignationStatus(
```

Comment:
This method does not belong to a generic helper. Please mode this to the service layer under the designation agreement scope.